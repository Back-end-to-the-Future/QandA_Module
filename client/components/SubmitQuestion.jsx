import React, { useState } from 'react';
import axios from 'axios';

import Button from 'react-bootstrap/Button';
import InputGroup from 'react-bootstrap/InputGroup';
import FormControl from 'react-bootstrap/FormControl';
import Modal from 'react-bootstrap/Modal';
import Alert from 'react-bootstrap/Alert';

const SubmitQuestion = (props) => {
  const [questionInput, setQuestionInput] = useState('');
  const [nicknameInput, setNicknameInput] = useState('');
  const [emailInput, setEmailInput] = useState('');
  const [validEntry, setValidEntry] = useState(true);

  const submitQuestion = () => {
    // isValidEmail is regex for anyString@anyString.anyString
    const isValidEmail = /\S+@\S+\.\S+/;
    let isValidBool = isValidEmail.test(emailInput);

    if (!isValidBool || questionInput === '' || nicknameInput === '') {
      setValidEntry(false);
    } else {
      let questionSubmission = {
        product_id: props.product_id,
        body: questionInput,
        name: nicknameInput,
        email: emailInput
      }

      axios.post('/question/add', questionSubmission);

      setValidEntry(true);
      props.handleCloseSubmit();
    }
  }

  return (
    <div className='submitQuestion'>
      <div className='submitQButton'>
        <Button variant="outline-dark" onClick={props.handleOpenSubmit}>
          Submit Question
        </Button>
      </div>

      <Modal show={props.show} onHide={props.handleCloseSubmit}>
        <Modal.Header closeButton>
          <Modal.Title id="contained-modal-title-vcenter">
            <h3>Ask Your Question</h3>
            <h5>About the {props.product_name}</h5>
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          {validEntry ? null : <Alert variant='danger' className='inValidAlert'>You must enter the following:</Alert>}
          <InputGroup className="mb-3">
            <InputGroup.Prepend>
              <InputGroup.Text className="inputLabel">*Your Question: </InputGroup.Text>
            </InputGroup.Prepend>
            <FormControl className='qBodyInput' as="textarea" maxLength={1000}
              onChange={(event) => setQuestionInput(event.target.value)}
            />
          </InputGroup>

          <InputGroup className="mb-3">
            <InputGroup.Prepend>
              <InputGroup.Text className="inputLabel">*What is your nickname: </InputGroup.Text>
            </InputGroup.Prepend>
            <FormControl className='qNameInput' type='text' maxLength={60}
              placeholder='Example: jackson11!'
              onChange={(event) => setNicknameInput(event.target.value)}
            />
            <div className='inputMessage'>
              <p>For privacy reasons, do not use your full name or email address.</p>
            </div>
          </InputGroup>

          <InputGroup className="mb-3">
            <InputGroup.Prepend>
              <InputGroup.Text className="inputLabel">*What is your email: </InputGroup.Text>
            </InputGroup.Prepend>
            <FormControl className='qEmailInput' type='text' maxLength={60}
              placeholder='Example: mwatney@mars.com'
              onChange={(event) => setEmailInput(event.target.value)}
            />
            <div className='inputMessage'>
              <p>For authentication reasons, you will not be emailed.</p>
            </div>
          </InputGroup>

        </Modal.Body>
        <Modal.Footer>
          <Button variant="outline-dark" className='saveButton' onClick={submitQuestion}>Save</Button>
        </Modal.Footer>
      </Modal>
    </div>
  );
}

export default SubmitQuestion;