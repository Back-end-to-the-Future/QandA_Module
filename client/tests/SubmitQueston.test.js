import React from 'react';
import SubmitQuestion from '../components/SubmitQuestion.jsx';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import EnzymeToJson from 'enzyme-to-json';
import { mount } from 'enzyme';

// Test Data for Submit Question
const product_name = 'Test Product Name';
const product_id = 0;
const mockHandleOpenSubmit = jest.fn();


describe('Answer Component', () => {
  it('should not close modal when input is invalid and save button is clicked', () => {

    const mockHandleCloseSubmit = jest.fn();

    const wrapper = shallow(
      <SubmitQuestion
        handleOpenSubmit={mockHandleOpenSubmit}
        handleCloseSubmit={mockHandleCloseSubmit}
        product_name={product_name}
        product_id={product_id}
        show={true}
      />
    );

    wrapper.find('.qBodyInput').simulate('change', { target: { value: 'Some Answer' } });
    wrapper.find('.qNameInput').simulate('change', { target: { value: 'Some Name' } });
    wrapper.find('.qEmailInput').simulate('change', { target: { value: 'Not an email' } });

    let saveButton = wrapper.find('.saveButton');
    saveButton.simulate('click');

    let alert = wrapper.find('.inValidAlert');

    expect(alert.length > 0).toBe(true);

    expect(mockHandleCloseSubmit.mock.calls.length).toBe(0);
  })

  it('should close modal and not show alert when input is valid and save button is clicked', () => {

    const mockHandleCloseSubmit = jest.fn();

    const wrapper = shallow(
      <SubmitQuestion
        handleOpenSubmit={mockHandleOpenSubmit}
        handleCloseSubmit={mockHandleCloseSubmit}
        product_name={product_name}
        product_id={product_id}
        show={true}
      />
    );

    wrapper.find('.qBodyInput').simulate('change', { target: { value: 'Some Question' } });
    wrapper.find('.qNameInput').simulate('change', { target: { value: 'Some Name' } });
    wrapper.find('.qEmailInput').simulate('change', { target: { value: 'some@email.com' } });

    let saveButton = wrapper.find('.saveButton');
    saveButton.simulate('click');

    let alert = wrapper.find('.inValidAlert');

    expect(alert.length > 0).toBe(false);

    expect(mockHandleCloseSubmit.mock.calls.length).toBe(1);
  })

  it('should match snapshot when show=true', () => {

    const mockHandleCloseSubmit = jest.fn();

    // this test did not like renderer for some reason
    const submitQuestionDisplay = mount(
      <SubmitQuestion
        handleOpenSubmit={mockHandleOpenSubmit}
        handleCloseSubmit={mockHandleCloseSubmit}
        product_name={product_name}
        product_id={product_id}
        show={true}
      />
    );

    expect(EnzymeToJson(submitQuestionDisplay)).toMatchSnapshot();

  })

});