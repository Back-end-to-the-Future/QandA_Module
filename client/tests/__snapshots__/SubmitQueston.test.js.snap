// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer Component should match snapshot when show=true 1`] = `
<SubmitQuestion
  handleCloseSubmit={[MockFunction]}
  handleOpenSubmit={[MockFunction]}
  product_id={0}
  product_name="Test Product Name"
  show={true}
>
  <div
    className="submitQuestion"
  >
    <div
      className="submitQButton"
    >
      <Button
        active={false}
        disabled={false}
        onClick={[MockFunction]}
        variant="outline-dark"
      >
        <button
          className="btn btn-outline-dark"
          disabled={false}
          onClick={[MockFunction]}
          type="button"
        >
          Submit Question
        </button>
      </Button>
    </div>
    <Modal
      animation={true}
      autoFocus={true}
      backdrop={true}
      dialogAs={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "ModalDialog",
          "render": [Function],
        }
      }
      enforceFocus={true}
      keyboard={true}
      onHide={[MockFunction]}
      restoreFocus={true}
      show={true}
    >
      <Modal
        autoFocus={true}
        backdrop={true}
        backdropTransition={[Function]}
        containerClassName="modal-open"
        enforceFocus={true}
        keyboard={true}
        manager={
          BootstrapModalManager {
            "containers": Array [
              <body
                class="modal-open"
                style="padding-right: 0px;"
              >
                <div
                  class="fade modal-backdrop show"
                />
                <div
                  aria-modal="true"
                  class="fade modal show"
                  role="dialog"
                  style="display: block;"
                  tabindex="-1"
                >
                  <div
                    class="modal-dialog"
                    role="document"
                  >
                    <div
                      class="modal-content"
                    >
                      <div
                        class="modal-header"
                      >
                        <div
                          class="modal-title h4"
                        >
                          <h3>
                            Ask Your Question
                          </h3>
                          <h5>
                            About the 
                            Test Product Name
                          </h5>
                        </div>
                        <button
                          class="close"
                          type="button"
                        >
                          <span
                            aria-hidden="true"
                          >
                            ×
                          </span>
                          <span
                            class="sr-only"
                          >
                            Close
                          </span>
                        </button>
                      </div>
                      <div
                        class="modal-body"
                      >
                        <div
                          class="mb-3 input-group"
                        >
                          <div
                            class="input-group-prepend"
                          >
                            <span
                              class="inputLabel input-group-text"
                            >
                              *Your Question: 
                            </span>
                          </div>
                          <textarea
                            class="qBodyInput form-control"
                            maxlength="1000"
                          />
                        </div>
                        <div
                          class="mb-3 input-group"
                        >
                          <div
                            class="input-group-prepend"
                          >
                            <span
                              class="inputLabel input-group-text"
                            >
                              *What is your nickname: 
                            </span>
                          </div>
                          <input
                            class="qNameInput form-control"
                            maxlength="60"
                            placeholder="Example: jackson11!"
                            type="text"
                          />
                        </div>
                        <p>
                          For privacy reasons, do not use your full name or email address.
                        </p>
                        <div
                          class="mb-3 input-group"
                        >
                          <div
                            class="input-group-prepend"
                          >
                            <span
                              class="inputLabel input-group-text"
                            >
                              *What is your email: 
                            </span>
                          </div>
                          <input
                            class="qEmailInput form-control"
                            maxlength="60"
                            placeholder="Example: mwatney@mars.com"
                            type="text"
                          />
                        </div>
                        <p>
                          For authentication reasons, you will not be emailed.
                        </p>
                      </div>
                      <div
                        class="modal-footer"
                      >
                        <button
                          class="saveButton btn btn-outline-dark"
                          type="button"
                        >
                          Save
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </body>,
            ],
            "data": Array [
              Object {
                "classes": Array [
                  "modal-open",
                ],
                "modals": Array [
                  Object {
                    "add": [Function],
                    "backdrop": <div
                      class="fade modal-backdrop show"
                    />,
                    "dialog": <div
                      aria-modal="true"
                      class="fade modal show"
                      role="dialog"
                      style="display: block;"
                      tabindex="-1"
                    >
                      <div
                        class="modal-dialog"
                        role="document"
                      >
                        <div
                          class="modal-content"
                        >
                          <div
                            class="modal-header"
                          >
                            <div
                              class="modal-title h4"
                            >
                              <h3>
                                Ask Your Question
                              </h3>
                              <h5>
                                About the 
                                Test Product Name
                              </h5>
                            </div>
                            <button
                              class="close"
                              type="button"
                            >
                              <span
                                aria-hidden="true"
                              >
                                ×
                              </span>
                              <span
                                class="sr-only"
                              >
                                Close
                              </span>
                            </button>
                          </div>
                          <div
                            class="modal-body"
                          >
                            <div
                              class="mb-3 input-group"
                            >
                              <div
                                class="input-group-prepend"
                              >
                                <span
                                  class="inputLabel input-group-text"
                                >
                                  *Your Question: 
                                </span>
                              </div>
                              <textarea
                                class="qBodyInput form-control"
                                maxlength="1000"
                              />
                            </div>
                            <div
                              class="mb-3 input-group"
                            >
                              <div
                                class="input-group-prepend"
                              >
                                <span
                                  class="inputLabel input-group-text"
                                >
                                  *What is your nickname: 
                                </span>
                              </div>
                              <input
                                class="qNameInput form-control"
                                maxlength="60"
                                placeholder="Example: jackson11!"
                                type="text"
                              />
                            </div>
                            <p>
                              For privacy reasons, do not use your full name or email address.
                            </p>
                            <div
                              class="mb-3 input-group"
                            >
                              <div
                                class="input-group-prepend"
                              >
                                <span
                                  class="inputLabel input-group-text"
                                >
                                  *What is your email: 
                                </span>
                              </div>
                              <input
                                class="qEmailInput form-control"
                                maxlength="60"
                                placeholder="Example: mwatney@mars.com"
                                type="text"
                              />
                            </div>
                            <p>
                              For authentication reasons, you will not be emailed.
                            </p>
                          </div>
                          <div
                            class="modal-footer"
                          >
                            <button
                              class="saveButton btn btn-outline-dark"
                              type="button"
                            >
                              Save
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>,
                    "isTopModal": [Function],
                    "remove": [Function],
                    "setBackdropRef": [Function],
                    "setDialogRef": [Function],
                  },
                ],
                "overflowing": true,
                "style": Object {
                  "overflow": "",
                  "paddingRight": "",
                },
              },
            ],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "modals": Array [
              Object {
                "add": [Function],
                "backdrop": <div
                  class="fade modal-backdrop show"
                />,
                "dialog": <div
                  aria-modal="true"
                  class="fade modal show"
                  role="dialog"
                  style="display: block;"
                  tabindex="-1"
                >
                  <div
                    class="modal-dialog"
                    role="document"
                  >
                    <div
                      class="modal-content"
                    >
                      <div
                        class="modal-header"
                      >
                        <div
                          class="modal-title h4"
                        >
                          <h3>
                            Ask Your Question
                          </h3>
                          <h5>
                            About the 
                            Test Product Name
                          </h5>
                        </div>
                        <button
                          class="close"
                          type="button"
                        >
                          <span
                            aria-hidden="true"
                          >
                            ×
                          </span>
                          <span
                            class="sr-only"
                          >
                            Close
                          </span>
                        </button>
                      </div>
                      <div
                        class="modal-body"
                      >
                        <div
                          class="mb-3 input-group"
                        >
                          <div
                            class="input-group-prepend"
                          >
                            <span
                              class="inputLabel input-group-text"
                            >
                              *Your Question: 
                            </span>
                          </div>
                          <textarea
                            class="qBodyInput form-control"
                            maxlength="1000"
                          />
                        </div>
                        <div
                          class="mb-3 input-group"
                        >
                          <div
                            class="input-group-prepend"
                          >
                            <span
                              class="inputLabel input-group-text"
                            >
                              *What is your nickname: 
                            </span>
                          </div>
                          <input
                            class="qNameInput form-control"
                            maxlength="60"
                            placeholder="Example: jackson11!"
                            type="text"
                          />
                        </div>
                        <p>
                          For privacy reasons, do not use your full name or email address.
                        </p>
                        <div
                          class="mb-3 input-group"
                        >
                          <div
                            class="input-group-prepend"
                          >
                            <span
                              class="inputLabel input-group-text"
                            >
                              *What is your email: 
                            </span>
                          </div>
                          <input
                            class="qEmailInput form-control"
                            maxlength="60"
                            placeholder="Example: mwatney@mars.com"
                            type="text"
                          />
                        </div>
                        <p>
                          For authentication reasons, you will not be emailed.
                        </p>
                      </div>
                      <div
                        class="modal-footer"
                      >
                        <button
                          class="saveButton btn btn-outline-dark"
                          type="button"
                        >
                          Save
                        </button>
                      </div>
                    </div>
                  </div>
                </div>,
                "isTopModal": [Function],
                "remove": [Function],
                "setBackdropRef": [Function],
                "setDialogRef": [Function],
              },
            ],
            "scrollbarSize": 0,
          }
        }
        onEnter={[Function]}
        onEntering={[Function]}
        onEscapeKeyDown={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onHide={[MockFunction]}
        renderBackdrop={[Function]}
        renderDialog={[Function]}
        restoreFocus={true}
        show={true}
        transition={[Function]}
      >
        <Portal
          containerInfo={
            <body
              class="modal-open"
              style="padding-right: 0px;"
            >
              <div
                class="fade modal-backdrop show"
              />
              <div
                aria-modal="true"
                class="fade modal show"
                role="dialog"
                style="display: block;"
                tabindex="-1"
              >
                <div
                  class="modal-dialog"
                  role="document"
                >
                  <div
                    class="modal-content"
                  >
                    <div
                      class="modal-header"
                    >
                      <div
                        class="modal-title h4"
                      >
                        <h3>
                          Ask Your Question
                        </h3>
                        <h5>
                          About the 
                          Test Product Name
                        </h5>
                      </div>
                      <button
                        class="close"
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          ×
                        </span>
                        <span
                          class="sr-only"
                        >
                          Close
                        </span>
                      </button>
                    </div>
                    <div
                      class="modal-body"
                    >
                      <div
                        class="mb-3 input-group"
                      >
                        <div
                          class="input-group-prepend"
                        >
                          <span
                            class="inputLabel input-group-text"
                          >
                            *Your Question: 
                          </span>
                        </div>
                        <textarea
                          class="qBodyInput form-control"
                          maxlength="1000"
                        />
                      </div>
                      <div
                        class="mb-3 input-group"
                      >
                        <div
                          class="input-group-prepend"
                        >
                          <span
                            class="inputLabel input-group-text"
                          >
                            *What is your nickname: 
                          </span>
                        </div>
                        <input
                          class="qNameInput form-control"
                          maxlength="60"
                          placeholder="Example: jackson11!"
                          type="text"
                        />
                      </div>
                      <p>
                        For privacy reasons, do not use your full name or email address.
                      </p>
                      <div
                        class="mb-3 input-group"
                      >
                        <div
                          class="input-group-prepend"
                        >
                          <span
                            class="inputLabel input-group-text"
                          >
                            *What is your email: 
                          </span>
                        </div>
                        <input
                          class="qEmailInput form-control"
                          maxlength="60"
                          placeholder="Example: mwatney@mars.com"
                          type="text"
                        />
                      </div>
                      <p>
                        For authentication reasons, you will not be emailed.
                      </p>
                    </div>
                    <div
                      class="modal-footer"
                    >
                      <button
                        class="saveButton btn btn-outline-dark"
                        type="button"
                      >
                        Save
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </body>
          }
        >
          <BackdropTransition
            appear={true}
            in={true}
          >
            <Fade
              appear={true}
              in={true}
              mountOnEnter={false}
              timeout={300}
              unmountOnExit={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={true}
                enter={true}
                exit={true}
                in={true}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={300}
                unmountOnExit={false}
              >
                <div
                  className="fade modal-backdrop show"
                  onClick={[Function]}
                />
              </Transition>
            </Fade>
          </BackdropTransition>
          <DialogTransition
            appear={true}
            in={true}
            onEnter={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            unmountOnExit={true}
          >
            <Fade
              appear={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              timeout={300}
              unmountOnExit={true}
            >
              <Transition
                addEndListener={[Function]}
                appear={true}
                enter={true}
                exit={true}
                in={true}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={300}
                unmountOnExit={true}
              >
                <div
                  aria-modal={true}
                  className="fade modal show"
                  onClick={[Function]}
                  onMouseUp={[Function]}
                  role="dialog"
                  style={
                    Object {
                      "paddingLeft": undefined,
                      "paddingRight": undefined,
                    }
                  }
                  tabIndex={-1}
                >
                  <ModalDialog
                    onMouseDown={[Function]}
                    role="document"
                  >
                    <div
                      className="modal-dialog"
                      onMouseDown={[Function]}
                      role="document"
                    >
                      <div
                        className="modal-content"
                      >
                        <ModalHeader
                          closeButton={true}
                          closeLabel="Close"
                        >
                          <div
                            className="modal-header"
                          >
                            <ModalTitle>
                              <ForwardRef
                                className="modal-title"
                              >
                                <div
                                  className="modal-title h4"
                                >
                                  <h3>
                                    Ask Your Question
                                  </h3>
                                  <h5>
                                    About the 
                                    Test Product Name
                                  </h5>
                                </div>
                              </ForwardRef>
                            </ModalTitle>
                            <CloseButton
                              label="Close"
                              onClick={[Function]}
                            >
                              <button
                                className="close"
                                onClick={[Function]}
                                type="button"
                              >
                                <span
                                  aria-hidden="true"
                                >
                                  ×
                                </span>
                                <span
                                  className="sr-only"
                                >
                                  Close
                                </span>
                              </button>
                            </CloseButton>
                          </div>
                        </ModalHeader>
                        <ModalBody>
                          <div
                            className="modal-body"
                          >
                            <InputGroup
                              className="mb-3"
                            >
                              <div
                                className="mb-3 input-group"
                              >
                                <InputGroupPrepend>
                                  <div
                                    className="input-group-prepend"
                                  >
                                    <InputGroupText
                                      className="inputLabel"
                                    >
                                      <span
                                        className="inputLabel input-group-text"
                                      >
                                        *Your Question: 
                                      </span>
                                    </InputGroupText>
                                  </div>
                                </InputGroupPrepend>
                                <FormControl
                                  as="textarea"
                                  className="qBodyInput"
                                  maxLength={1000}
                                  onChange={[Function]}
                                >
                                  <textarea
                                    className="qBodyInput form-control"
                                    maxLength={1000}
                                    onChange={[Function]}
                                  />
                                </FormControl>
                              </div>
                            </InputGroup>
                            <InputGroup
                              className="mb-3"
                            >
                              <div
                                className="mb-3 input-group"
                              >
                                <InputGroupPrepend>
                                  <div
                                    className="input-group-prepend"
                                  >
                                    <InputGroupText
                                      className="inputLabel"
                                    >
                                      <span
                                        className="inputLabel input-group-text"
                                      >
                                        *What is your nickname: 
                                      </span>
                                    </InputGroupText>
                                  </div>
                                </InputGroupPrepend>
                                <FormControl
                                  className="qNameInput"
                                  maxLength={60}
                                  onChange={[Function]}
                                  placeholder="Example: jackson11!"
                                  type="text"
                                >
                                  <input
                                    className="qNameInput form-control"
                                    maxLength={60}
                                    onChange={[Function]}
                                    placeholder="Example: jackson11!"
                                    type="text"
                                  />
                                </FormControl>
                              </div>
                            </InputGroup>
                            <p>
                              For privacy reasons, do not use your full name or email address.
                            </p>
                            <InputGroup
                              className="mb-3"
                            >
                              <div
                                className="mb-3 input-group"
                              >
                                <InputGroupPrepend>
                                  <div
                                    className="input-group-prepend"
                                  >
                                    <InputGroupText
                                      className="inputLabel"
                                    >
                                      <span
                                        className="inputLabel input-group-text"
                                      >
                                        *What is your email: 
                                      </span>
                                    </InputGroupText>
                                  </div>
                                </InputGroupPrepend>
                                <FormControl
                                  className="qEmailInput"
                                  maxLength={60}
                                  onChange={[Function]}
                                  placeholder="Example: mwatney@mars.com"
                                  type="text"
                                >
                                  <input
                                    className="qEmailInput form-control"
                                    maxLength={60}
                                    onChange={[Function]}
                                    placeholder="Example: mwatney@mars.com"
                                    type="text"
                                  />
                                </FormControl>
                              </div>
                            </InputGroup>
                            <p>
                              For authentication reasons, you will not be emailed.
                            </p>
                          </div>
                        </ModalBody>
                        <ModalFooter>
                          <div
                            className="modal-footer"
                          >
                            <Button
                              active={false}
                              className="saveButton"
                              disabled={false}
                              onClick={[Function]}
                              variant="outline-dark"
                            >
                              <button
                                className="saveButton btn btn-outline-dark"
                                disabled={false}
                                onClick={[Function]}
                                type="button"
                              >
                                Save
                              </button>
                            </Button>
                          </div>
                        </ModalFooter>
                      </div>
                    </div>
                  </ModalDialog>
                </div>
              </Transition>
            </Fade>
          </DialogTransition>
        </Portal>
      </Modal>
    </Modal>
  </div>
</SubmitQuestion>
`;
